src/main/java/com/iota/iri/Iota.java

```diff
@@ -40,90 +39,48 @@
-        double alpha = configuration.doubling(Configuration.DefaultConfSettings.TIPSELECTION_ALPHA.name());
-        int belowMaxDepthTxLimit = configuration.integer(
-                Configuration.DefaultConfSettings.BELOW_MAX_DEPTH_TRANSACTION_LIMIT);
-
-        boolean dontValidateMilestoneSig = configuration.booling(Configuration.DefaultConfSettings
-                .DONT_VALIDATE_TESTNET_MILESTONE_SIG);
-        int transactionPacketSize = configuration.integer(Configuration.DefaultConfSettings.TRANSACTION_PACKET_SIZE);
-
-        maxTipSearchDepth = configuration.integer(Configuration.DefaultConfSettings.MAX_DEPTH);
-        if(testnet) {
-            String coordinatorTrytes = configuration.string(Configuration.DefaultConfSettings.COORDINATOR);
-            if(StringUtils.isNotEmpty(coordinatorTrytes)) {
-                coordinator = new Hash(coordinatorTrytes);
-            } else {
-                log.warn("No coordinator address given for testnet. Defaulting to "
-                        + Configuration.TESTNET_COORDINATOR_ADDRESS);
-                coordinator = new Hash(Configuration.TESTNET_COORDINATOR_ADDRESS);
-            }
-        } else {
-            coordinator = new Hash(Configuration.MAINNET_COORDINATOR_ADDRESS);
-        }
+        Snapshot initialSnapshot = Snapshot.init(configuration.getSnapshotFile(),
+                configuration.getSnapshotSignatureFile(), configuration.isTestnet()).clone();
         tangle = new Tangle();
-        messageQ = new MessageQ(configuration.integer(Configuration.DefaultConfSettings.ZMQ_PORT),
-                configuration.string(Configuration.DefaultConfSettings.ZMQ_IPC),
-                configuration.integer(Configuration.DefaultConfSettings.ZMQ_THREADS),
-                configuration.booling(Configuration.DefaultConfSettings.ZMQ_ENABLED)
-                );
+        messageQ = MessageQ.createWith(configuration);
         tipsViewModel = new TipsViewModel();
         transactionRequester = new TransactionRequester(tangle, messageQ);
         transactionValidator = new TransactionValidator(tangle, tipsViewModel, transactionRequester, messageQ,
-                snapshotTimestamp);
-        milestone = new Milestone(tangle, coordinator, initialSnapshot, transactionValidator, testnet, messageQ,
-                numKeysMilestone, milestoneStartIndex, dontValidateMilestoneSig);
-        node = new Node(configuration, tangle, transactionValidator, transactionRequester, tipsViewModel, milestone, messageQ);
-        replicator = new Replicator(node, tcpPort, maxPeers, testnet, transactionPacketSize);
-        udpReceiver = new UDPReceiver(udpPort, node, configuration.integer(Configuration.DefaultConfSettings.TRANSACTION_PACKET_SIZE));
+                configuration);
+        milestone = new Milestone(tangle, transactionValidator, messageQ, configuration, initialSnapshot);
+        node = new Node(configuration, tangle, transactionValidator, transactionRequester, tipsViewModel, milestone,
```

Comment:
We should be consistent in the location of the `configuration` argument. Either beginning or end?